publish.go

```diff
@@ -207,6 +292,15 @@ func (publisher *Publisher) Publish(
 		message.Body = data
 		message.Headers = tableToAMQPTable(options.Headers)
 		message.Expiration = options.Expiration
+		message.ContentEncoding = options.ContentEncoding
```

Comment:
I really like that you're adding these options! Thanks.

Are we setting "sane" defaults for all of these new options though? Just want to double check your thoughts.